webpack/helpers.js

```diff
@@ -125,13 +128,63 @@ function convertJsonToSassGeneral(data) {
 	return output;
 }
 
+/**
+ * Convert Recursive Json data to Css variables.
+ *
+ * @param {object} data Json data to convert.
+ */
+function convertJsonToSassVariables(data) {
+	let output = '';
+
+	for (const property in data) {
+		if (Object.prototype.hasOwnProperty.call(data, property)) {
+			switch (typeof data[property]) {
+				case 'object':
+					for (const inner in data[property]) {
+						if (Object.prototype.hasOwnProperty.call(data[property], inner)) {
+							switch (property) {
+								case 'colors':
+									output += `--${property}-${data[property][inner].slug}: ${data[property][inner].color};\n`;
+									break;
+								case 'gradient':
+									output += `--${property}-${data[property][inner].slug}: ${data[property][inner].gradient};\n`;
+									break;
+								case 'fontSizes':
+									output += `--${property}-${data[property][inner].slug}: ${data[property][inner].slug};\n`;
+									break;
+								default:
+									output += `--${property}-${inner}: ${data[property][inner]};\n`;
+									break;
+							}
+						}
+					}
+					break;
+				default:
+					output += `--${property}: ${data[property]};\n`;
+					break;
+			}
+		}
+	}
```

Comment:
There isn't any prettier way to write this? Like some helper method or something. Loop then if, then switch, then another loop... Looks messy :S

Reply:
This could be way more elegant with a simple if else:

```js
if (typeof data[property] !== 'object') {
    output += `--${property}: ${data[property]};\n`;
    break;
}

for (const inner ...)
...
```

Also, calling `Object.prototype.hasOwnProperty` could be avoided with just going through the `Object.entries` in a `for ... of ...` loop, because then you're sure that the object has that property.

Removes tons of indentation 😄